### PR TITLE
Documentation updates, part II

### PR DESCRIPTION
**Pull request by welchr**
Might want to wait on merging this in since I may make additional changes.

### CODE REVIEW

File: src/scoring.js
Comment (abought) [inline src/scoring.js]: Given the simplicity of this method, is there a strong reason to couple it specifically and only to Bayes factors? It seems that other scoring methods might also have a need to be normalized- leaving room for a codebase to grow is a delicate balance.

(I realize that naming the method `posteriorProbabilities` probably undercuts my case, but, well, that's irony for you...)

Anyway, let me know when the PR and changes are ready. I have a few more edits to docs and examples and will hold off pushing them upstream. See personal repo:
https://github.com/abought/gwas-credible-sets/commit/17465c17463c8a017a80900bca93ddf8c5599d35

Comment (welchr) [reply src/scoring.js]: Actually you could just push your changes and I'll rebase on top of them, I don't think I will likely change much more, everything looks great. :) 

It's a simple method but unfortunately it only really makes sense to convert Bayes factors to posterior probabilities (and it's actually specific to our method due to the assumptions made for that to be the case, as I understand it.) 

Maybe a more generic function could be substituted entirely. This function is basically [normalize](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.normalize.html) using L1-norm (we sum the values, L1-norm sums the absolute values, but all of our values are always positive anyway.)

Comment (abought) [reply src/scoring.js]: > Actually you could just push your changes and I'll rebase on top of them, I don't think I will likely change much more, everything looks great. :)

Done. 

Only other pending change: I'm separating the `findCredibleSet` method out from the marking options (boolean, scaled, etc).  Eg: `markBoolean(credibleSet)`

It turns out there is a good use case for marking / visualizing the data in multiple ways on a plot, and it seemed silly to keep doing the underlying credible set calc every time.

**Comment by abought** [comment]
I think I've made all the changes I planned on to this; feel free to rebase. Once we're both comfortable with the docs and API in this repo, I'll tag a version, and push to NPM as part of broader release prep.

**Comment by abought** [comment]
No other changes currently planned on my end, aside from a final readthrough of docs- care to rebase and prep this PR for merge?

My goal is to consolidate this (and have a tagged NPM version pushed) by Wednesday, in tandem with the credible sets example in LocusZoom. One more feature pending for the annotation track, but otherwise I'm currently in the polish and unit testing / review phase on both sides. I can move sooner or later depending on how formal a release we want to push out.

**Comment by welchr** [comment]
I added a little more description of the method under `docs/method/`, which was the easy part. 

The other changes are a little more complicated. You may want to check out this branch specifically locally to see what changed before merging in. 

I made an attempt to get the paths for everything to work correctly whether viewed on github or viewed from the `docs/` folder, which was the main issue. But I don't think it quite worked out for the timings link. The symbolic link doesn't resolve for some reason (though it seems to work fine in another test repo of mine, so I'm not sure what the difference is here.) Don't know of a solution other than just moving the file under `docs/` instead of `test/timings/`.

**Comment by welchr** [comment]
Should have said - if you're happy with these changes, go ahead and merge/rebase in, I think this should do it from my end. You should also have commit access to this branch if you want to make any changes and then rebase this back onto develop/master. 

When you've got things all patched and ready on Weds, we could send to Mike/Goncalo for a final look before shipping it off, probably on Thurs or Friday.

File: docs/README.md
Comment (abought) [inline docs/README.md:1]: Small question: since JSDoc automatically builds index.html using a specified README.md, what is the advantage of moving the file into this folder?

(Github automatically renders a top-level README with the repo, and jsdoc is more agnostic about where the README lives)

File: docs/method/render.sh
Comment (abought) [inline docs/method/render.sh:2]: This description of the method is a big, and readable, improvement- thanks!

If you think this will get revised in the future, it could conceivably be added as an npm script so that all maintenance operations for the repo can be found and run from one consistent place.

File: docs/tests/timings/profile-method.js
Comment (abought) [inline docs/tests/timings/profile-method.js:1]: I may clean this up a bit- we've set Github pages to serve assets from the `docs` folder of the master branch, and web servers don't like to serve things from arbitrary locations on the hard drive.

As a general rule going forward, I'll try to encourage that the `docs` folder be a destination for built assets- something that we can wipe completely if need be and regenerate. (this is a crude but effective means of pruning dead assets, which can creep in whenever a build or template system is involved). 

Accordingly, I may move the pandoc files and scripts to another folder as part of post-merge cleanup.

**Review by abought — APPROVED** [review]

File: docs/README.md
Comment (welchr) [reply docs/README.md:1]: I was mainly trying to get everything to view correctly whether you were looking at the HTML (via github pages) or looking at the README (via github directly). The links weren't quite working out because README.md had paths relative to itself, but then the index.html under docs/ would not have paths updated accordingly. 

Any solution is good with me really, just need to double check all the links to make sure they work.

File: docs/method/render.sh
Comment (welchr) [reply docs/method/render.sh:2]: Could it be included in the npm docs command? I'm not sure everyone will have pandoc installed on their machine, I guess is the only issue, and npm probably can't pull it in (I'm assuming?)

File: docs/tests/timings/profile-method.js
Comment (welchr) [reply docs/tests/timings/profile-method.js:1]: Makes good sense to me.

File: docs/README.md
Comment (abought) [reply docs/README.md:1]: > Any solution is good with me really, just need to double check all the links to make sure they work.

In the end, I think the underlying problem is that we're trying to accommodate at least three separate viewing modes simultaneously. For now I've hard-coded a few GitHub-Pages specific links (arbitrarily picking the website as the primary way people read the docs, and not trying too hard to support the others). Changes pushed:
https://github.com/statgen/gwas-credible-sets/commit/8dec8c5c3cd21317c237a6d4bd9baa838d0e52c4

It seems silly, but things that get rebuilt often- like the `docs` folder- are a recipe for merge conflicts, so keeping that folder simple helps avoid headaches down the line. 

Very open to revisiting making things more elegant post-release, if you think people would prefer to read docs in other than HTML format.

File: docs/method/render.sh
Comment (abought) [reply docs/method/render.sh:2]: In theory NPM can run any command on your system, not just node modules. 

I've added a "user beware" message to the README about the missing dependencies- since it also requires a latex install, I avoided even trying to impose assumptions on the user. :)

**Pull request merged**
